[21.01] Release Notes for 21.01

mvdbeek opened the pull request.

mvdbeek committed:
Prepare release notes

jmchilton committed:
Some work on 21.01 release notes.

jmchilton committed:
More progress on 21.01 release notes.

jmchilton committed:
More progress on release notes for 21.01.

jmchilton committed:
More 21.01 release notes work.

jmchilton committed:
Some more progress on 21.01 release notes.

jmchilton committed:
More progress on 21.01 release notes.

hexylena committed:
starting user facing notes

jdavcs commented on doc/source/releases/21.01.rst:
```diff
@@ -0,0 +1,683 @@
+* Add FastAPI as backend and start porting backend to an ASGI webapp to
+  enable modern programming features such a message queues and web sockets.
+  `Pull Request 10823`_, `Pull Request 11074`_
+*  Replace StreamBall, replace tarballs with zip
```
```suggestion
* Replace StreamBall, replace tarballs with zip
```
I think that's what's causing it to be interpreted as a block quote.

mvdbeek replied:
Thanks, fixed it.

hexylena commented on doc/source/releases/21.01_announce_user.rst:
```diff
@@ -0,0 +1,135 @@
+* Add author and license metadata and corresponding web microdata annotations for tools
+  and workflows.
+  `Pull Request 10408`_
+
```
todo @hexylena add workflow invocation progress thing for @mvdbeek . thanks for the reminder of sth I missed!

hexylena replied:
Did not intend to forget this -- sometimes just can't tell if something is interesting from the PR titles / multiple minor PRs to enhance something get lost sometimes :(

please let me know if I'm misisng other stuff!

hexylena committed:
Write release notes

hexylena committed:
newlines whatever

hexylena commented on doc/source/releases/21.01_announce_user.rst:
```diff
@@ -0,0 +1,148 @@
+* Allow the Apply Rules tool to attach tags to output datasets.
+  `Pull Request 10382`_
+* allow empty file uploads
+  (thanks to `@galaxyproject <https://github.com/galaxyproject>`__).
```
This looks like a bug? 11079 is bernt-matthias

hexylena committed:
Correct pr

hexylena committed:
Spellcheck

hexylena committed:
Done

mvdbeek committed:
Update merged PRs

mvdbeek committed:
Add additional PRs

mvdbeek committed:
Fix leading space

mvdbeek committed:
Skip PRs that are already in the release notes

When updating the (already edited) release notes during the PR process
it is a major pain which PRs are really new. This makes that much
easier. Just be sure to keep the anchors (`.. bug`, `.. enhancement`) in until we are ready to
publish the release

mvdbeek committed:
Tweak release issue template

mvdbeek committed:
Write 21.01_announce.rst, add 21.01_announce_user to toc

mvdbeek committed:
Mention tool document cache

mvdbeek committed:
Fix link

mvdbeek committed:
Add latest set of 21.01 PRs

mvdbeek committed:
Release testing team credits

mvdbeek committed:
Latest set of PRs

mvdbeek committed:
Add release testing credits to template

dannon committed:
Minor fixups to release notes

mvdbeek committed:
Make release team a list

dannon committed:
Visualis|zation standardization

dannon committed:
Fix keith's last name

dannon committed:
Minor fix of duplicate punctuation

OlegZharkov commented on doc/source/releases/21.01.rst:
```diff
@@ -0,0 +1,878 @@
+  `Pull Request 11533`_
+* Add input step data label when extracting workflow.
+  `Pull Request 11532`_
+* Fix saving FileSource parameter in a workflow.
```
it's 20.09 fix. Do we also include it in 21.01 release notes?

mvdbeek replied:
Yes, otherwise it is never listed in any release notes.

OlegZharkov replied:
> Yes, otherwise it is never listed in any release notes.

Ok. I see

dannon committed:
Fix its it's

dannon committed:
Minor rewording and standardizations

jdavcs commented on doc/source/releases/21.01_announce.rst:
```diff
@@ -1,11 +1,84 @@
+- `John Chilton <https://github.com/jmchilton>`_
+- `Keith Suderman <https://github.com/ksuderman>`_
+- `Luke Sargent <https://github.com/luke-c-sargent>`_
+- `Sergey Golitsynskiy <https://github.com/ic4f>`_
```
```suggestion
- `Assunta DeSanto <https://github.com/assuntad23>`_
- `David López <https://github.com/davelopez>`_
- `John Chilton <https://github.com/jmchilton>`_
- `Keith Suderman <https://github.com/ksuderman>`_
- `Sergey Golitsynskiy <https://github.com/ic4f>`_
```
Jay and Luke will be included in the same section in the 20.09 release notes.

mvdbeek committed:
Reduce to current release testing team

Co-authored-by: Sergey Golitsynskiy <sgolitsynskiy@gmail.com>

jdavcs commented on doc/source/releases/21.01_announce.rst:
```diff
@@ -1,11 +1,82 @@
+  This method is more efficient and can create large archives without affecting the Galaxy server process. The required setup is described in the admin documentation for `proxying Galaxy with NGINX <https://docs.galaxyproject.org/en/release_21.01/admin/nginx.html#creating-archives-with-mod-zip>`__.
+
+**Workflow Enhancements and Fixes**
+  This release contains many new features, enhancements and bug fixes for creating, running and reporting workflows described in greater details in the `21.01 user release notes <21.01_announce_user.html>`__
```
```suggestion
  This release contains many new features, enhancements and bug fixes for creating, running and reporting workflows described in greater detail in the `21.01 user release notes <21.01_announce_user.html>`__
```

jdavcs commented on doc/source/releases/21.01_announce.rst:
```diff
@@ -1,11 +1,82 @@
+
+**History Import and Export**
+  Histories can now be imported from and exported to the pluggable file source architecture introduced in the 20.09 release of Galaxy.
+  This includes the users' FTP folder, S3, dropbox, owncloud etc. In addition the History Export and Import dialogues have been rewritten
```
```suggestion
  This includes the users' FTP folder, S3, Dropbox, ownCloud etc. In addition the History Export and Import dialogues have been rewritten
```

mvdbeek committed:
Fix service names

Co-authored-by: Sergey Golitsynskiy <sgolitsynskiy@gmail.com>

jdavcs commented on doc/source/releases/21.01_announce.rst:
```diff
@@ -1,11 +1,82 @@
+===========================================================
+
+**Deprecation of support for Python 3.5**
+  Galaxy requires Python 3.6 or newer. Instruction for updating Python can be found in the `Supported Python versions <https://docs.galaxyproject.org/en/release_21.01/admin/python.html#supported-python-versions>`__ section of the documentation.
```
```suggestion
  Galaxy requires Python 3.6 or newer. Instructions for updating Python can be found in the `Supported Python versions <https://docs.galaxyproject.org/en/release_21.01/admin/python.html#supported-python-versions>`__ section of the documentation.
```

mvdbeek committed:
Fix greater details -> greater detail

Co-authored-by: Sergey Golitsynskiy <sgolitsynskiy@gmail.com>

mvdbeek committed:
Fix typo

Co-authored-by: Sergey Golitsynskiy <sgolitsynskiy@gmail.com>

jdavcs commented on doc/source/releases/21.01_announce.rst:
```diff
@@ -1,11 +1,82 @@
+  Note that if you update to Python 3.7.2 or newer you need to set py-call-osafterfork to ``false`` in the uwsgi section of galaxy.yml.
+
+**Removal of IUC conda channel from default configuration**
+  The IUC conda channel has been removed from the default configuration. This channel was used for installing dependencies that were old and not available in bioconda or conda-forge. These dependencies are now available in Bioconda. By removing this channel Conda can work more efficiently and ignore outdated dependencies.
```
```suggestion
  The IUC conda channel has been removed from the default configuration. This channel was used for installing dependencies that were old and not available in bioconda or conda-forge. These dependencies are now available in bioconda. By removing this channel Conda can work more efficiently and ignore outdated dependencies.
```
I think in both cases "bioconda" should be lowercased as it refers to the name of a channel. I think uppercasing "Conda" in the second sentence is correct.

jdavcs commented on doc/source/releases/21.01_announce.rst:
```diff
@@ -1,11 +1,82 @@
+
+**Removal of IUC conda channel from default configuration**
+  The IUC conda channel has been removed from the default configuration. This channel was used for installing dependencies that were old and not available in bioconda or conda-forge. These dependencies are now available in Bioconda. By removing this channel Conda can work more efficiently and ignore outdated dependencies.
+  If you have specified alternative channels in your Galaxy configuration we recommend remving the iuc channel from the `conda_ensure_channels <https://github.com/galaxyproject/galaxy/blob/283ec1772fcc0e574997dc2c5ad12dc86e3e1d37/lib/galaxy/config/sample/galaxy.yml.sample#L324>`__ variable.
```
```suggestion
  If you have specified alternative channels in your Galaxy configuration we recommend removing the iuc channel from the `conda_ensure_channels <https://github.com/galaxyproject/galaxy/blob/283ec1772fcc0e574997dc2c5ad12dc86e3e1d37/lib/galaxy/config/sample/galaxy.yml.sample#L324>`__ variable.
```

mvdbeek committed:
Lowercase channel, fix typo

Co-authored-by: Sergey Golitsynskiy <sgolitsynskiy@gmail.com>

jdavcs commented on doc/source/releases/21.01_announce.rst:
```diff
@@ -1,11 +1,82 @@
+  If you have specified alternative channels in your Galaxy configuration we recommend remving the iuc channel from the `conda_ensure_channels <https://github.com/galaxyproject/galaxy/blob/283ec1772fcc0e574997dc2c5ad12dc86e3e1d37/lib/galaxy/config/sample/galaxy.yml.sample#L324>`__ variable.
+
+**Removal of the communication server**
+  This feature had not seen greater adoption and was now removed. Most communication has moved to Gitter or other chat services.
```
```suggestion
  This feature has not seen greater adoption and has been removed. Most communication has moved to Gitter or other chat services.
```
(or "...had not seen... over the past n years...")

dannon committed:
Update doc/source/releases/21.01_announce.rst

Co-authored-by: Sergey Golitsynskiy <sgolitsynskiy@gmail.com>

jdavcs commented on doc/source/releases/21.01_announce.rst:
```diff
@@ -1,11 +1,82 @@
+
+      $ git fetch origin && git checkout release_21.01 && git pull --ff-only origin release_21.01
+
+See the `community hub <https://galaxyproject.org/develop/source-code/>`__ for additional details regarding the source code locations.
```
```suggestion
See the `community hub <https://galaxyproject.org/develop/source-code/>`__ for additional details on source code locations.
```

mvdbeek replied:
Can you also update the template in scripts/bootstrap_history.py ?

jdavcs replied:
Done.

jdavcs commented on doc/source/releases/21.01_announce_user.rst:
```diff
@@ -0,0 +1,181 @@
+
+   <iframe width="560" height="315" src="https://www.youtube-nocookie.com/embed/TmZzfaKf1V0" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+
+If you're building advanced workflows utilizing Galaxy's powerful sub-workflows for re-usable workflow components, then you'll be pleased to know you can now automatically update those to the latest version
```
```suggestion
If you're building advanced workflows utilizing Galaxy's powerful sub-workflows for reusable workflow components, then you'll be pleased to know you can now automatically update those to the latest version
```

mvdbeek committed:
re-usable -> reuseable

Co-authored-by: Sergey Golitsynskiy <sgolitsynskiy@gmail.com>

jdavcs committed:
Update wording

jdavcs reviewed (approved):
Looks great! 
(I've proofread everything, I hope)
